projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/notebook.py

```diff
@@ -0,0 +1,120 @@
+            self.file_path = file_path
+
+            content = json.loads(file_path.read_text())
+            cells = [
+                cell
+                for cell in content["cells"]
+                if cell["cell_type"] == "code"
+                and "vdk" in cell["metadata"].get("tags", {})
+            ]
+            for cell in cells:
+                code = "".join(cell["source"])
+                if code.startswith("%sql") or "def run(" in code:
+                    if code.startswith("%sql"):
+                        code = "".join(cell["source"])
+                        code.replace(";", "")
+                        self.sql_and_run_cells.append(
+                            [code.replace("%sql", ""), TYPE_SQL]
+                        )
+                    else:
+                        self.sql_and_run_cells.append(
+                            ["".join(cell["source"]), TYPE_PYTHON]
+                        )
+                else:
+                    self.python_helper_cells.append("".join(cell["source"]))
+
+        except json.JSONDecodeError as e:
+            errors.log_and_rethrow(
+                to_be_fixed_by=errors.ResolvableBy.USER_ERROR,
+                log=log,
+                what_happened=f"Failed to read the {file_path.name} file.",
+                why_it_happened=f"The provided {file_path.name} cannot be loaded into json format and "
+                f"cannot be read as a Jupyter notebook",
+                consequences=errors.MSG_CONSEQUENCE_TERMINATING_APP,
+                countermeasures=f"Check the {file_path.name} format again",
+                exception=e,
+                wrap_in_vdk_error=True,
+            )
+
+    def extract_notebook_steps(self, context: JobContext):
+        for index, cell in enumerate(self.sql_and_run_cells):
+            if cell[1] == TYPE_SQL:
+                step = NotebookStep(
+                    name="".join(
+                        [
+                            self.file_path.name.replace(".ipynb", "_"),
+                            str(index),
+                        ]
+                    ),
+                    type=TYPE_SQL,
+                    runner_func=NotebookStepFuncFactory.run_sql_step,
+                    file_path=self.file_path,
+                    job_dir=context.job_directory,
+                    code=cell[0],
+                )
+            else:
+                step = NotebookStep(
+                    name="".join(
+                        [
+                            self.file_path.name.replace(".ipynb", "_"),
+                            str(index),
+                        ]
+                    ),
+                    type=TYPE_PYTHON,
+                    runner_func=NotebookStepFuncFactory.run_python_step,
+                    file_path=self.file_path,
+                    job_dir=context.job_directory,
+                    code="\n".join(self.python_helper_cells) + "\n" + cell[0],
+                )
```

Comment:
<img width="1354" alt="Screenshot 2022-12-08 at 10 28 03" src="https://user-images.githubusercontent.com/11415233/206396895-9fd818fd-8c69-4dc5-bde7-17f0dca242a2.png">
nit: take a look how similar if-else contents are; maybe deduplicate code?

Reply:
There were some problems with the types, for some reason. When I didn't use if else but directly use variable for the code and the type instead, I got platform error, bc vdk was trying to run the python step as an sql one